### DISCUSSION TITLE
Termux -> ngork authtoken

### DISCUSSION THREAD

**Discussion by XTrim07** [termux/termux-app#4066] [Q&A] [answered]
Hi! I want to run ngok on my android device (no root). So, I installed this package
` wget https://bin.equinox.io/c/4VmDzA7iaHb/ngrok-stable-linux-arm64.tgz `
and unzipped. Everything seemed to be going well, but when tried to add an authtoken this error occurred. What should I do and what could be the causes?

```
./ngrok authtoken <token>
error: "/data/data/com.termux/files/home/ngrok" has unexpected e_type: 2
```

**Comment by TomJo2000** [comment, marked as answer]
Ngrok is a proprietary network tunneling application, as such we can offer neither a package, nor support for it.
You can find the documentation for it at:
https://ngrok.com/docs/

Or see this list for open source alternatives:
https://github.com/anderspitman/awesome-tunneling

**Comment by Alipro123789** [comment]
Brother, did you find a solution to this problem?

**Comment by PepeTux** [comment]
If you want to run ngrok on termux, you must install the termux version from F-Droid:
https://f-droid.org/en/packages/com.termux/

the 	**suggested**  is Version 0.118.1 (1000), use that, 
![imagen](https://github.com/user-attachments/assets/223ccd42-42c8-40dc-b8c2-8ad5705f6f5c)

I had the same error:
error: "/data/data/com.termux/files/home/ngrok" has unexpected e_type: 2
runing ngrok from the play store termux version will not work.

After installing that version from F-Droid and download ngrok from the official web it works:
https://download.ngrok.com/linux?tab=download

![imagen](https://github.com/user-attachments/assets/87a93ef8-6dab-41e6-8e38-084be12376bb)

Now everything is working !

Maybe you want follow this guide:
https://dev.to/fazilchengapra/how-to-install-ngrok-in-termux-a-step-by-step-guide-4dnk

Running on Android 11, 13, arm64-v8a.
![imagen](https://github.com/user-attachments/assets/72a530b9-a3be-41c5-bfba-524bb23d112e)


Saludos
JL

**Reply by sahilrajput03** [reply]
This worked. Thanks.
